### DISCUSSION TITLE
Keydelay

### DISCUSSION THREAD

**Discussion by mdeberts** [autokey/autokey#563] [Q&A]
Hey,

can anyone tell me what's the standard key delay for autokey?
Like autohotkey for window has a standard key delay of ~10 ms.
Are keysends here send at 0 ms or 10 ms?

**Comment by josephj11** [comment]
Maybe one of the developers can respond, but based on observation, I would say it's 0. Lots of scripts end up having to add multiple delays between events to allow the receiving applications to keep up without losing characters, etc. I even wrote a type_slow() function that emits one character at a time with a programmable delay after each one for extreme cases.

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.

**Reply by ineuw** [reply]
I've done extensive timing and delay related tests with Autokey and there is 0 delay. ` time.sleep(0.1)`  is the minimum required for sending keystrokes or storing and retrieving data from the clipboard. The keyboard's navigation keys are especially slow and need to be addressed. Python is just as fast as Autohotkey.  Data manipulation in memory is instantaneous. But, placing and retrieving requires delay. The attached file is a .py file renamed to .txt. I wrote some notes in the script.

[alt-o.txt](https://github.com/autokey/autokey/files/6689678/alt-o.txt)

**Reply by josephj11** [reply]
@ineuw Thanks.

FYI I have opened #566 suggesting a new Paste using option for phrases targeting really slow applications.
